src/PythonMigrationViewExtension/Differ/IDiffViewViewModel.cs

```diff
@@ -1,9 +1,17 @@
 ﻿namespace Dynamo.PythonMigration.Differ
 {
+    public enum State
+    {
+        NoChanges,
+        HasChanges,
+        Error
+    }
+
     public interface IDiffViewViewModel
     {
         ViewMode ViewMode { get; }
-        bool HasChanges { get; }
+        State DiffState { get; }
+        bool Error { get; set; }
```

Comment:
Do we need this boolean if `DiffState` can already be `Error`?

Reply:
I'm using the `DiffState` instead of the previously used `HasChanges` boolean in the xaml view, which is why I needed to change it to an enum from a bool as it now represents 3 states - `HasChanges` - when the diff viewer shows code diff, `NoChanges` - when we show the message saying the code has no changes, and this new `Error` state, where I wanted to show the error message in the same UI.

Reply:
I understand that but my question remains. Why do we need the boolean `Error`?

Reply:
I'm using it as a flag to set it to `true` after catching the `PythonException`. I could probably just use the `DiffState` enum and use that instead. Let me try.